components/metrics/metrics_commitments.go

```diff
@@ -0,0 +1,99 @@
+package metrics
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/iotaledger/hive.go/runtime/event"
+	"github.com/iotaledger/iota-core/components/metrics/collector"
+	"github.com/iotaledger/iota-core/pkg/protocol/chainmanager"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/notarization"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+const (
+	commitmentsNamespace = "commitments"
+
+	latestCommitment    = "latest"
+	finalizedCommitment = "finalized"
+	forksCount          = "forks_total"
+	missingRequested    = "missing_requested_total"
+	missingReceived     = "missing_received_total"
+	acceptedBlocks      = "accepted_blocks"
+	transactions        = "accepted_transactions"
+	validators          = "active_validators"
+)
+
+var CommitmentsMetrics = collector.NewCollection(commitmentsNamespace,
+	collector.WithMetric(collector.NewMetric(latestCommitment,
+		collector.WithType(collector.Gauge),
+		collector.WithHelp("Last commitment of the node."),
+		collector.WithLabels("commitment"),
+		collector.WithPruningDelay(10*time.Minute),
+		collector.WithInitFunc(func() {
+			deps.Protocol.Events.Engine.Notarization.SlotCommitted.Hook(func(details *notarization.SlotCommittedDetails) {
+				deps.Collector.Update(commitmentsNamespace, latestCommitment, float64(details.Commitment.ID().Index()), details.Commitment.ID().String())
+			}, event.WithWorkerPool(Component.WorkerPool))
+		}),
+	)),
+	collector.WithMetric(collector.NewMetric(finalizedCommitment,
+		collector.WithType(collector.Gauge),
+		collector.WithHelp("Last commitment finalized by the node."),
+		collector.WithInitFunc(func() {
+			deps.Protocol.Events.Engine.SlotGadget.SlotFinalized.Hook(func(slot iotago.SlotIndex) {
+				deps.Collector.Update(commitmentsNamespace, finalizedCommitment, float64(slot))
```

Comment:
The latest finalized slot must always match the corresponding commitment published as part of the latestCommitment metric with label, and we don't really specifically consume the commitment ID in the dashboard exacly because of such assumption: that's why I didn't add a labelled vector here.